src/arithmetic.js

```diff
@@ -129,26 +152,35 @@ function processQueue(queue) {
 
 const noWhitespace = /\s/g;
 
+// Variables are not replaced if the next non-whitespace character is a '('
+// e.g. max(5 + max)
+// the first max would not be replaced as it is a function not a variable
+const buildVariableReplacer = (key) => new RegExp(`${key}(?!\\s*\\()`, 'g');
+
 export function runArithmetic(formulaText, values = {}) {
   // first replace variables with their actual values
   // (we do this here rather than treating the variable names as tokens,
   // so that the tokeniser doesn't get confused by variable names with
   // u and x in them)
   let valuedText = formulaText;
-  Object.entries(values).map(([key, value]) => {
+  Object.entries(values).forEach(([key, value]) => {
     if (Number.isNaN(parseFloat(value))) {
       throw new Error('Invalid value substitution');
     }
 
-    valuedText = valuedText.replace(new RegExp(key, 'g'), value);
+    valuedText = valuedText.replace(buildVariableReplacer(key), value);
   });
 
   // strip out all whitespace
   const strippedText = valuedText.replace(noWhitespace, '');
 
+  if (strippedText.match(/([(,],)|(,\))/g)) throw new Error('Leading or trailing comma detected');
+  // make sure that functions are case insensitive
+  const lowercaseText = strippedText.toLowerCase();
```

Comment:
Will replacement of variables containing uppercase letters still work?

Reply:
To my knowledge, all replacement is completed by this point, so this shouldn't break anything. (line 166 to 172).

Reply:
Nice! Could you add a unit tests for uppercase vars to capture this in code?

Reply:
Good idea!